app/src/screen.c

```diff
@@ -65,6 +65,33 @@ static void set_window_size(struct screen *screen, struct size new_size) {
     }
 }
 
+void get_window_scale(SDL_Window *window, float *scale_x, float *scale_y) {
+    int win_w, win_h;
+    SDL_GetWindowSize(window, &win_w, &win_h);
+    int output_w, output_h;
+    SDL_GL_GetDrawableSize(window, &output_w, &output_h);
+    *scale_x = (float)output_w / win_w;
+    *scale_y = (float)output_h / win_h;
+}
+
+SDL_bool screen_update_logical_size_if_needed(struct screen *screen) {
+    float scale_x, scale_y;
+    get_window_scale(screen->window, &scale_x, &scale_y);
+    if (scale_x != screen->scale_x || scale_y != screen->scale_y) {
```

Comment:
I agree that comparing float values for equality is fragile.

Updating logical size when the frame size changes is ok, but updating on `SDL_WINDOWEVENT_SIZE_CHANGED` is unnecessary.

Reply:
OK, then we can keep the last width and height passed to `SDL_RenderSetLogicalSize()` (instead of the scale), and call it only when the parameters have changed.